src/ai.hpp

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+#include "car.hpp"
+#include "direction.hpp"
+#include "lane.hpp"
+#include <cmath> 
+#include <cstdlib>
+#include <ctime>
+#include <limits>
+#include <map>
+#include <set>
+#include <stdexcept>
+#include <vector>
+
+struct PlayerOutcome
+{
+	float cars_turned_per_move;
+	direction most_villainous_direction;
+};
+
+class AI
+{
+public:
+	const int MaxPly = 3;
```

Comment:
We'll probably need to change this number after testing. Hopefully it can go higher.